src/coresyftools/coresyftools/gpt_tool.py

```diff
@@ -0,0 +1,101 @@
+'''Defines a CoReSyF Tool consisting of a SNAP single gpt operation.'''
+import os
+import subprocess
+import re
+from tool import CoReSyFTool
+from manifest import InvalidManifestException
+
+class TooManyInputArgumentsException(Exception):
+    pass
+
+class ToManyOutputArgumentsException(Exception):
+    pass
+
+class GPTExecutionException(Exception):
+    '''Error occurred during a SNAP gpt execution'''
+
+    ERROR_REGEX = re.compile('Error:')
+
+    def __init__(self, returncode, msg):
+        self.returncode = returncode
+        self.errors = [line[7:] for line in msg.split(
+            os.linesep) if self.ERROR_REGEX.match(line)]
+        super(GPTExecutionException, self).__init__(self.errors)
+
+
+class GPTGraphFileNotFound(Exception):
+    pass
+
+
+class GPTCoReSyFTool(CoReSyFTool):
+    '''CoReSyF Tool consisting of a single SNAP gpt operation.'''
+
+    DEFAULT_FORMAT = 'GeoTIFF-BigTIFF'
+    DEFAULT_EXT = 'tif'
+    DEFAULT_GPT_GRAPH_FILE_NAME = 'gpt_graph.xml'
+
+    def _validate_operation(self, operation):
+        if 'operation' not in operation and ('graph' not in operation or not operation['graph']):
+            raise InvalidManifestException('A operation or graph flag should be present.')
+        if 'operation' in operation and 'graph' in operation and operation['graph']:
+            raise InvalidManifestException('Can not be operation and graph at same time.')
+        if 'graph' in self.operation and self.operation['graph']:
+            graph_file = os.path.join(self.context_directory, self.DEFAULT_GPT_GRAPH_FILE_NAME)
+            if not os.path.exists(graph_file):
+                raise GPTGraphFileNotFound(graph_file)
+
+    def run(self, bindings):
+        if len(self.arg_parser.inputs) > 1:
+            raise TooManyInputArgumentsException()
+        if len(self.arg_parser.outputs) > 1:
+            raise TooManyInputArgumentsException()
+
+        bindings = bindings.copy()
+        operator = None
+        if 'operation' in self.operation:
+            operator = self.operation['operation']
+        elif 'graph' in self.operation and self.operation['graph']:
+            graph_file = os.path.join(self.context_directory, self.DEFAULT_GPT_GRAPH_FILE_NAME)
+            if not os.path.exists(graph_file):
+                raise GPTGraphFileNotFound(graph_file)
+            operator = graph_file
+        if 'parameters' in self.operation:
```

Comment:
I would extract this to a different function `get_parameters`

Reply:
What would that function encapsulate? If you had to write a docstring, what would you write? I do not see any independent _concept_ that would be meaningful outside this function. For me this is an atomic step for the `run` operation.

Reply:
Simplified to:
```python

 if 'parameters' in self.operation:
            bindings.update(self.operation['parameters'])
```

Reply:
Agreed.